src/world.c

```diff
@@ -15,9 +22,18 @@ void create_world(int p, int q, world_func func, void *arg) {
             float f = simplex2(x * 0.01, z * 0.01, 4, 0.5, 2);
             float g = simplex2(-x * 0.01, -z * 0.01, 2, 0.9, 2);
             int mh = g * 32 + 16;
-            int h = f * mh;
+            int h;
+            if(x > 0 && x < 2000 && z < 0 && z < 2000){
```

Comment:
See pull comment. I believe this line is your issue. (I believe it should be z > 0 && z < 2000 )